src/game/game.js

```diff
@@ -422,8 +422,7 @@
                 get() {
 
                     try {
-                        runCodeCache[userId].memory._parsed = JSON.parse(runCodeCache[userId].memory.get() || "{}");
-                        runCodeCache[userId].memory._parsed.__proto__ = null;
+                        runCodeCache[userId].memory._parsed = _sandboxedEval('JSON.parse')(runCodeCache[userId].memory.get() || "{}");
```

Comment:
The closure-scoped value `_sandboxedEval` should not be directly used in any callbacks below, since it will lead to inconsistent behavior when multiple users are being executed in the same process. Instead, you should add it to `runCodeCache[userId]` object above and execute it as `runCodeCache[userId].sandboxedEval('JSON.parse')`.